common/flatpak-instance.c

```diff
@@ -725,9 +987,67 @@ flatpak_instance_gc_per_app_dirs (const char *instance_id,
   if (statbuf.st_mtime + 3 >= time (NULL))
     return glnx_throw (error, "lock file too recent, avoiding race condition");
 
-  /* Clean up the app's /tmp, which could contain relatively large files */
+  /* /dev/shm is offloaded onto the host's /dev/shm to get consistent
+   * free space behaviour and make sure it's actually in RAM. It could
+   * contain relatively large files, so we clean it up.
+   *
+   * In principle this could be used for other directories such as /tmp,
+   * in a loop over an array of paths (hence this indentation), but we
+   * only do this for /dev/shm right now. */
+    {
```

Comment:
Clang complains about this bracket because is right after the `if ()`:
```
common/flatpak-instance.c:997:5: error: misleading indentation; statement is not part of the previous 'if' [-Werror,-Wmisleading-indentation]
    {
    ^
```

Reply:
I was hoping the leading block- comment would make it consider the indentation to be non-misleading, like gcc does.

For now, I can add `G_STMT_START`/`G_STMT_END` or `if (1)` or `do`/`while (0)` to silence that warning.

The way this block will end up depends on the maintainers' answer to the "Design queries" in the PR description. If the maintainers want both `/tmp` and `/dev/shm` handled like this, then this block will become a real loop. Conversely, if the maintainers want exactly one directory handled like this, then I could unindent the block and move the local variables to the top (but I'd rather not do that until the design queries are answered).

Reply:
... or if the maintainers want `/dev/shm` handled like `/tmp` is now, then this block (and the code that supports it) will vanish entirely.

Reply:
My clang 11 doesn't complain about this, but I've added `do/while(0)` for now. When the maintainers have decided how they want `/tmp` and `/dev/shm` handled, I'll reformat this.